### PR TITLE
Remove fastcrypto: take 2

### PR DESCRIPTION
**Pull request by ljedrz**
Applies on top of `fix/integration` and supersedes https://github.com/eqlabs/bullshark-bft/pull/37. Compiles, but haven't run the tests yet. I moved the changes from the former PR and applied some new ones where applicable.

Also, there's a change I didn't understand in the former PR - the `narwhal.yaml` one, so be extra vigilant about that bit.

### CODE REVIEW

File: narwhal/crypto/src/network_keypair.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    pub fn private(&self) -> &SecretKey {
+        &self.0.secret
+    }
+
+    pub fn copy(&self) -> Self {
+        Self(ed25519::Keypair::from_bytes(&self.0.to_bytes()).unwrap())
+    }
+
+    pub fn encode_base64(&self) -> String {
```
Comment (niklaslong) [inline narwhal/crypto/src/network_keypair.rs:98]: There's an `EncodeDecodeBase64` trait in the module, perhaps these could be moved to an impl of that. We did it initially to preserve some of the key APIs when changing the crypto.

File: narwhal/node/tests/staged/narwhal.yaml [hunk 2]
```diff
@@ -85,10 +85,7 @@ WorkerIndex:
         TYPENAME: WorkerInfo
 WorkerInfo:
   STRUCT:
-    - name:
-        TUPLEARRAY:
-          CONTENT: U8
-          SIZE: 32
+    - name: STR
```
Comment (niklaslong) [inline narwhal/node/tests/staged/narwhal.yaml:88]: I think this is expected, the underlying structure for the `NetworkPublicKey`, which is the `name` field of the `WorkerInfo` struct changed. @joske might have more insights to provide.

**Comment by niklaslong** [comment]
I've managed to reproduce the CI failure locally; not sure what's up with `test_determinism`.

File: narwhal/node/tests/staged/narwhal.yaml (hunk 2 above)
Comment (joske) [reply narwhal/node/tests/staged/narwhal.yaml:88]: yes this was already in the original PR

**Review by joske — APPROVED** [review]
I wanted to do this tonight, but you already did :-D

**Pull request merged**